routing/endpointregistry_test.go

```diff
@@ -101,6 +114,48 @@ func TestDoRemovesOldEntries(t *testing.T) {
 	assert.Equal(t, int64(0), mRemoved.InflightRequests())
 }
 
+func TestRaceReadWrite(t *testing.T) {
+	r := routing.NewEndpointRegistry(routing.RegistryOptions{})
+	duration := time.Second
+
+	wg := sync.WaitGroup{}
+	wg.Add(2)
+	go func() {
+		defer wg.Done()
+		for start := time.Now(); time.Since(start) < duration; {
+			r.GetMetrics("some key")
+		}
+	}()
+	go func() {
+		defer wg.Done()
+		for start := time.Now(); time.Since(start) < duration; {
+			r.GetMetrics("some key").IncInflightRequest()
+		}
+	}()
+	wg.Wait()
```

Comment:
This can be also done with timer so it will not need to call time.Since in the loop, see https://github.com/zalando/skipper/blob/5f706e5058720a70ecd3427ebe99306a4c918bc9/loadbalancer/fadein_test.go#L93-L107